packages/babel-helper-define-polyfill-provider/src/utils.ts

```diff
@@ -25,11 +21,9 @@ function resolveId(path): string {
     return path.node.name;
   }
 
-  if (path.isPure()) {
-    const { deopt } = path.evaluate();
-    if (deopt && deopt.isIdentifier()) {
-      return deopt.node.name;
-    }
+  const resolved = path.resolve();
+  if (resolved.isIdentifier()) {
+    return resolved.node.name;
```

Comment:
While working on https://github.com/babel/babel-polyfills/pull/225#pullrequestreview-2652557253 I realized there is a observable difference.
```
var b = Math;
b.f16round();
```
Now it won't return `Math`, maybe I should open a PR to restore this behavior?

Reply:
Yes, please!